src/main/java/io/github/ibuildthecloud/gdapi/validation/ValidationHandler.java

```diff
@@ -338,11 +345,10 @@ protected Object checkType(String fieldName, Object value, Class<?> type) {
             result.addAll(items);
             return result;
         }
-
         FieldType type = subTypes.get(0);
         for ( Object item : items ) {
-            item = convert(fieldName, null, type, subTypes.subList(1, subTypes.size()),
-                    subTypesNames.subList(1, subTypesNames.size()), item, context);
+            item = convert(fieldName, null, type, subTypes.subList(0, subTypes.size()),
```

Comment:
1 was correct.  Did this fix a bug?  The reason is this code is supposed to support nesting so you can have a array[map[string]].  The first item in the subtype list needs to be popped because the item in array[map[string]] is map[string].  So we drop the first type.